cps/tasks/metadata_extract.py

```diff
@@ -160,8 +160,14 @@ def run(self, worker_thread):
             self._remove_shorts_from_db(conn)
             requested_urls = self._fetch_requested_urls(conn)
             if not requested_urls:
-                self.message = f"{self.media_url_link} failed: Video not available."
-                conn.execute("DELETE FROM media WHERE path = ?", (self.media_url,))
+                if self.unavailable:
+                    self.message = f"{self.media_url_link} failed: Video not available."
+                elif error_message := conn.execute("SELECT error FROM media WHERE ? LIKE '%' || extractor_id || '%'", (self.media_url,)).fetchone()[0]:
```

Comment:
I don't quite understand why we're taking the first error where the extractor_id matches the media_url.

Couldn't we take all records (or the first record) where `webpath = media_url` and error is not empty?

```python
errors = conn.execute(
    "SELECT error FROM media WHERE webpath = ? AND error != ''", (self.media_url,))
    .fetchone()
```

And we could also get the id column in one query:

```python
(id, error) = conn.execute(
    "SELECT id, error FROM media WHERE webpath = ? AND error != ''", (self.media_url,))
    .fetchone()
```

Reply:
Good point. However, due to SQLite UNIQUE constraint, we don't expect 2 relevant records. And this record should be deleted right after to allow the user to force redownload. If ever another record with the same extractor ID is present, it will never match because its `webpath` would have a timestamp appended to it and its error column would not be NULL.